thrift/src/main/java/com/intellij/plugins/thrift/completion/ThriftGotoDeclarationHandler.java

```diff
@@ -0,0 +1,92 @@
+package com.intellij.plugins.thrift.completion;
+
+import com.intellij.codeInsight.navigation.actions.GotoDeclarationHandler;
+import com.intellij.openapi.actionSystem.DataContext;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.project.Project;
+import com.intellij.plugins.thrift.index.ThriftDeclarationIndex;
+import com.intellij.plugins.thrift.lang.psi.ThriftDeclaration;
+import com.intellij.psi.*;
+import com.intellij.psi.search.GlobalSearchScope;
+import org.apache.commons.compress.utils.Lists;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.plugins.scala.lang.psi.api.base.ScReference;
+import org.jetbrains.plugins.scala.lang.resolve.ScalaResolveResult;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ThriftGotoDeclarationHandler implements GotoDeclarationHandler {
+
+    @NotNull
+    private List<ThriftDeclaration> getThriftDeclarations(List<String> fullyQualifiedNames, Project project) {
+        try {
+            return fullyQualifiedNames.stream()
+                    .filter(name -> getElementName(name) != null && getPackageName(name) != null)
+                    .map(name -> ThriftDeclarationIndex.findDeclaration(getElementName(name), getPackageName(name), project, GlobalSearchScope.allScope(project)))
+                    .flatMap(Collection::stream)
+                    .collect(Collectors.toList());
+        } catch (NoClassDefFoundError ignored) {}
```

Comment:
I am curious when does this exception happen

Reply:
Now I am curious too - I didn't think about this `try.. catch` block by myself (this try..catch block come from `GoToThriftDefinition#getThriftDeclarations`) Now I don't see any place where it could be thrown